Makefile

```diff
@@ -1,19 +1,26 @@
 #!/usr/bin/make
 
+TOX_VER = $(shell tox --version 2>/dev/null || echo "0.0")
+TOX_VER_MAJOR := $(shell echo $(TOX_VER) | cut -f1 -d.)
+TOX_VER_MINOR := $(shell echo $(TOX_VER) | cut -f2 -d.)
+TOX_GT_1_8 := $(shell [ $(TOX_VER_MAJOR) -gt 1 -o \( $(TOX_VER_MAJOR) -eq 1 -a $(TOX_VER_MINOR) -ge 8 \) ] && echo true)
+
 all: lint unit_test
 
 
 .PHONY: clean
 clean:
 	@rm -rf .tox
 
-.PHONY: apt_prereqs
-apt_prereqs:
-	@# Need tox, but don't install the apt version unless we have to (don't want to conflict with pip)
-	@which tox >/dev/null || (sudo apt-get install -y python-pip && sudo pip install tox)
+.PHONY: tox_prereqs
+tox_prereqs:
+ifneq ($(TOX_GT_1_8),true)
+        @echo "Pip installing tox"
+        sudo apt-get install -y python-pip && sudo pip install -U 'tox>=1.8'
```

Comment:
Very many tests assume sudo powers and apt-get or pip install things.  Tests are containerized already using charmbox which is what this is assuming.  Maybe we should just fix charmbox instead, but then we'd still already have the pip behavior that's in there currently.